internal/file/digest.go

```diff
@@ -0,0 +1,35 @@
+package file
+
+import (
+	"crypto/sha1" //nolint:gosec
+	"encoding/hex"
+	"io"
+	"os"
+
+	"golang.org/x/xerrors"
+
+	"github.com/anchore/syft/internal/log"
+)
+
+const DefaultDigestAlgorithm = "sha1"
+
+// Digest takes a filepath and returns a sha1 digest for the given contents
+func Digest(filepath string) (digest string, err error) {
+	file, err := os.Open(filepath)
+	if err != nil {
+		return digest, xerrors.Errorf("unable to open file: %s - %w", filepath, err)
```

Comment:
Just curious, why `xerrors.Errorf` instead of `fmt.Errorf`? (ditto for a few lines down from here, too)

Reply:
I think this was a mistake on my part. I thought by including this I would get more functionality from the `%w` verb, but on further inspection, I think everything I want for error wrapping is already in the std library.